2019-12-naming-part1/naming.md

```diff
@@ -0,0 +1,234 @@
+
+In terms of the Open Closed Principle, the code is more open to the change. For example, a new requirement to add a salutation would only necessitate changes to implementation internals, no API-breaking name change would be necessary. By increasing the level of abstraction, you have created an approprate and resilient function name.
+
+**GUIDELINE: Name functions one level of abstraction higher than their implementation**
+
+### Climb higher: ask a friend for stability
+
+When you seriously consider function names, it's not uncommon to come up with several candidates that seem equally viable. In those cases, break the tie by choosing words that your customers and product team use. In other words, when you're not sure, ask others for help. Just as you might ask a friend to stabilize the ladder while you climb that one rung higher to reach the work you need to do, asking for help in naming has a stabilizing effect on code.
+
+Application lexicons arise from conversations. It reduces mental friction when the vocabulary of your team matches the words you see on the screen in your text editor. Without having to translate person-speak to code-speak, you remove just one more bit of mental processing you'd do best to avoid. Agreeing on names reduces the overall organizational cost of writing software. In this spirit, it can be helpful, especially for newcomers on your team, to maintain a glossary of terms for your application.
+
+**GUIDELINE: Choose names from your application lexicon**
+
+## Climb up and down the ladder
+
+Does software design really demand this level of attention to detail, this careful weighing of names for something as unremarkable and ordinary as a single function? If you want to do your best work, then yes, it does. Names matter. Attention to detail matters. Writing software is about more than just getting something to work. At an abstract level, it's an expression of thought and communication, like writing, like speaking, like relationship-building. Your thoughtfulness around names will pay off.
+
+A literal function name indicates that you know what a function does but may not have considered what it means or how it's perceived by others who use it. Poor names betray a lack of understanding. When your internal and external dialog spans multiple levels of the Abstraction Ladder, you become open to the full breadth of understanding that categorical thinking bestows. That deeper understanding leads to better code, in every way.
+
+The guidelines in this article are by no means an exhaustive list. Naming presents infinite considerations and unique circumstances. There is no one-size-fits-all list of advice. Yet, you have some guidelines for improving names and your code will be better for considering them. Thinking more broadly, how do you help others improve? How do you spread value throughout your organization? Part 2 of this article will explore those questions and offer advice about naming classes and modules.
```

Comment:
How about saying, "Part 2 of this article, coming in the next blog post, will explore . . ."

Reply:
I'm not sure that will be the next blog post. :)